pkg/authorization/authorizer/scope/authorizer.go

```diff
@@ -0,0 +1,71 @@
+package scope
+
+import (
+	"fmt"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	defaultauthorizer "github.com/openshift/origin/pkg/authorization/authorizer"
+	"github.com/openshift/origin/pkg/authorization/rulevalidation"
+)
+
+type scopeAuthorizer struct {
+	delegate            defaultauthorizer.Authorizer
+	clusterPolicyGetter rulevalidation.ClusterPolicyGetter
+}
+
+func NewAuthorizer(delegate defaultauthorizer.Authorizer, clusterPolicyGetter rulevalidation.ClusterPolicyGetter) defaultauthorizer.Authorizer {
+	return &scopeAuthorizer{delegate: delegate, clusterPolicyGetter: clusterPolicyGetter}
+}
+
+func (a *scopeAuthorizer) Authorize(ctx kapi.Context, passedAttributes defaultauthorizer.AuthorizationAttributes) (bool, string, error) {
+	user, exists := kapi.UserFrom(ctx)
+	if !exists {
+		return false, "", fmt.Errorf("no user")
```

Comment:
Suggest a more descriptive error message here if there's any risk of it ending up at an API caller.